pkg/seedadmissioncontroller/webhooks/admission/extensionresources/admission.go

```diff
@@ -0,0 +1,253 @@
+				new := n.(*extensionsv1alpha1.Worker)
+				return validation.ValidateWorker(new)
+			},
+			validateResourceUpdate: func(n, o client.Object) field.ErrorList {
+				new := n.(*extensionsv1alpha1.Worker)
+				old := o.(*extensionsv1alpha1.Worker)
+				return validation.ValidateWorkerUpdate(new, old)
+			}})
+
+	return &h
+}
+
+type handler struct {
+	reader    client.Reader
+	decoder   *admission.Decoder
+	logger    logr.Logger
+	artifacts map[metav1.GroupVersionResource]artifact
+}
+
+func (h *handler) addResource(gvr metav1.GroupVersionResource, art artifact) {
+	h.artifacts[gvr] = art
+}
+
+var _ admission.Handler = &handler{}
+
+func (h *handler) InjectAPIReader(reader client.Reader) error {
+	h.reader = reader
+	return nil
+}
+
+func (h *handler) InjectDecoder(d *admission.Decoder) error {
+	h.decoder = d
+	return nil
+}
+
+func (h *handler) Handle(ctx context.Context, ar admission.Request) admission.Response {
+	_, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	artifact, ok := h.artifacts[ar.Resource]
+	if !ok {
+		return admission.Allowed("validation not found for the given resource")
+	}
+
+	switch ar.Operation {
+	case v1.Create:
+		return h.handleValidation(ar.Object, ar.OldObject, artifact.newEntity, artifact.validateResource)
+	case v1.Update:
+		return h.handleValidation(ar.Object, ar.OldObject, artifact.newEntity, artifact.validateResourceUpdate)
+	default:
+		return admission.Allowed("operation is not CREATE or UPDATE")
+	}
+}
+
+type newEntity func() client.Object
+type validate func(new, old client.Object) field.ErrorList
+
+// artifact servers as a helper to setup the corresponding function.
+type artifact struct {
+	// newEntity is a simple function that creates and returns a new resource.
+	newEntity newEntity
+
+	// validateResource is a wrapper function for the different create validation functions.
+	validateResource validate
+
+	// validateResourceUpdate is a wrapper function for the different update validation functions.
+	validateResourceUpdate validate
+}
+
+func (h handler) handleValidation(object, oldObject runtime.RawExtension, newEntity newEntity, validate validate) admission.Response {
+	obj := newEntity()
+	if err := h.decoder.DecodeRaw(object, obj); err != nil {
+		h.logger.Error(err, "could not decode ar", "ar", object)
+		return admission.Errored(http.StatusUnprocessableEntity, fmt.Errorf("could not decode ar %v: %w", object, err))
+	}
+
+	oldObj := newEntity()
+	if len(oldObject.Raw) != 0 {
+		if err := h.decoder.DecodeRaw(oldObject, oldObj); err != nil {
+			h.logger.Error(err, "could not decode old object", "object", oldObj)
+			return admission.Errored(http.StatusBadRequest, fmt.Errorf("could not decode old object %v: %v", oldObj, err))
+		}
+	}
+
+	errors := validate(obj, oldObj)
+	if len(errors) == 0 {
+		return admission.Allowed("validation successful")
+	}
+
+	var err = apierrors.NewInvalid(gvk.GroupKind(), "", errors)
+
+	return admission.Denied(err.Error())
```

Comment:
Can you reverse the logic here such as?

```suggestion
	if len(errors) != 0 {
		var err = apierrors.NewInvalid(gvk.GroupKind(), "", errors)
		return admission.Denied(err.Error())
	}

	return admission.Allowed("validation successful")
```